lib/wallet/src/Cardano/Wallet/Shelley/Network/Node.hs

```diff
@@ -539,7 +549,12 @@ withNodeNetworkLayerBase
             => RetryHandlers
             -> IO
                 ( STM IO (Tip (CardanoBlock StandardCrypto))
-                , TMVar IO (W.ProtocolParameters, W.SlottingParameters)
+                , TMVar
+                    IO
+                    ( W.ProtocolParameters
+                    , W.SlottingParameters
+                    , MaybeInRecentEra Write.ProtocolParameters
+                    )
```

Comment:
Perhaps it would be worth defining a record type to replace this tuple?

Something like:
```hs
data NetworkParams = NetworkParams
    { protocolParams :: MaybeInRecentEra Write.ProtocolParameters
    , protocolParamsLegacy :: W.ProtocolParameters
    , slottingParamsLegacy :: W.SlottingParameters
    }
```
One benefit is that we could access each field with an ordinary field accessor, rather than having to use positional lenses.

Reply:
Agreed

Reply:
Another option would be:

```Haskell
- | Contains all the information we fetch over LSQ from the local node tip
data LocalNodeTipState = LocalNodeTipState
    { protocolParams :: MaybeInRecentEra Write.ProtocolParameters
    , protocolParamsLegacy :: W.ProtocolParameters 
    , slottingParamsLegacy :: W.SlottingParameters 
    , era :: _
    , timeInterpreter :: _
    }
```

which if it works out would have the benefit of:
- Not creating further variants of the `PParams` type
- Simplifying the internals of `NetworkLayer` / making sense conceptually

This would be done by merging these callbacks into one:
https://github.com/cardano-foundation/cardano-wallet/blob/e9f88b3084d96801e572d9bb9feb84096036e0e9/lib/wallet/src/Cardano/Wallet/Shelley/Network/Node.hs#L912-L914